models/Asset/Document.php

```diff
@@ -163,6 +165,49 @@ public function getText(int $page = null): ?string
         return null;
     }
 
+    public function checkIfPdfContainsJS(): bool
+    {
+        if (!$this->isPdfScanningEnabled()) {
+           return false;
+        }
+
+        $this->setCustomSetting(
+            self::CUSTOM_SETTING_PDF_SCAN_STATUS,
+            Model\Asset\Enum\PdfScanStatus::IN_PROGRESS->value
+        );
+
+        $chunkSize = 1024;
+        $filePointer = $this->getStream();
+
+        $tagLength = strlen('/JS');
+
+        while ($chunk = fread($filePointer, $chunkSize)) {
+            if (strlen($chunk) <= $tagLength) {
+                break;
+            }
+
+            if (str_contains($chunk, '/JS') || str_contains($chunk, '/JavaScript')) {
+                $this->setCustomSetting(
+                    self::CUSTOM_SETTING_PDF_SCAN_STATUS,
+                    Model\Asset\Enum\PdfScanStatus::UNSAFE->value
+                );
+                return true;
+            }
+        }
+
+        $this->setCustomSetting(
+            self::CUSTOM_SETTING_PDF_SCAN_STATUS,
+            Model\Asset\Enum\PdfScanStatus::SAFE->value
+        );
+
+        return true;
+    }
+
+    public function getScanStatus(): ?string
+    {
+        return $this->getCustomSetting(self::CUSTOM_SETTING_PDF_SCAN_STATUS);
+    }
```

Comment:
What about returning the enum here?

```suggestion
    public function getScanStatus(): ?Model\Asset\Enum\PdfScanStatus
    {
        return Model\Asset\Enum\PdfScanStatus::tryFrom($this->getCustomSetting(self::CUSTOM_SETTING_PDF_SCAN_STATUS));
    }
```

Reply:
At the moment, it is called in twig templates. Do you know an easy way to read these values in twig?

Reply:
Oh, I see. I've no experience with enums in Twig. Maybe this helps: https://github.com/twigphp/Twig/issues/3681#issuecomment-1162728959

Reply:
Ok. It simply works with accessing the property `.value` so let's merge it.